drivers/scheduler/rke/rke.go

```diff
@@ -1,28 +1,270 @@
+	"io/ioutil"
+	"net/http"
+	"strings"
 )
 
 const (
-	// SchedName is the name of the kubernetes scheduler driver implementation
-	SchedName = "rke"
-	// SystemdSchedServiceName is the name of the system service resposible for scheduling
-	SystemdSchedServiceName = "kubelet"
+	// scheduleName is the name of the kubernetes scheduler driver implementation
+	scheduleName = "rke"
+	// SystemdScheduleServiceName is the name of the system service responsible for scheduling
+	SystemdScheduleServiceName = "kubelet"
+	// PxLabelNameKey is key for map
+	PxLabelNameKey = "name"
+	// PxLabelValue portworx pod label
+	PxLabelValue = "portworx"
 )
 
-type rke struct {
+var RancherClusterParametersValue *RancherClusterParameters
+
+type Rancher struct {
 	kube.K8s
+	client *rancherClient.Client
+}
+
+type RancherClusterParameters struct {
+	Token     string
+	Endpoint  string
+	AccessKey string
+	SecretKey string
+}
+
+// String returns the string name of this driver.
+func (r *Rancher) String() string {
+	return scheduleName
+}
+
+// Init Initialize the driver
+func (r *Rancher) Init(scheduleOpts scheduler.InitOptions) error {
+	var err error
+	err = r.K8s.Init(scheduleOpts)
+	RancherClusterParametersValue, err = r.GetRancherClusterParametersValue()
+	if err != nil {
+		return err
+	}
+	rancherClientOpts := rancherClientBase.ClientOpts{
+		URL:       RancherClusterParametersValue.Endpoint,
+		TokenKey:  RancherClusterParametersValue.Token,
+		AccessKey: RancherClusterParametersValue.AccessKey,
+		SecretKey: RancherClusterParametersValue.SecretKey,
+		Insecure:  true,
+	}
+	r.client, err = rancherClient.NewClient(&rancherClientOpts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetRancherClusterParametersValue returns the rancher token, endpoint, secret key, access key
+func (r *Rancher) GetRancherClusterParametersValue() (*RancherClusterParameters, error) {
+	var rkeData map[string]interface{}
+	var rkeParameters RancherClusterParameters
+	// TODO Rancher URL for cloud cluster will not be fetched from master node IP
+	masterNodeName := node.GetMasterNodes()[0].Name
+	endpoint := "https://" + masterNodeName + "/v3"
+	rancherURL := "https://" + masterNodeName + "/v3-public/localProviders/local?action=login"
+	//TODO: Get the values from config map after https://github.com/portworx/torpedo/pull/1517 is merged
+	username := "admin"
+	password := "1JVWD8juKPB8fVgK"
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+	body := strings.NewReader(fmt.Sprintf(`{"username":"%s","password":"%s"}`, username, password))
+	req, err := http.NewRequest("POST", rancherURL, body)
+	if err != nil {
+		log.Errorf("Failed to create rancher POST request:", err)
+		return &rkeParameters, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Infof("Failed to send request:", err)
```

Comment:
How about using either log.Errorf or log.Infof consistently for handling all errors within this function?

Reply:
done